### PR TITLE
Use new port router for onchanupgradetry

### PR DESCRIPTION
**Pull request by chatton**
## Description

**Marking as R4R for E2Es**

closes: #7027

This PR refactors fee middleware, ICA and callbacks to use the new port router.

Some wiring changes in app.gos needed to be fixed also.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/05-port/types/ibc_legacy_module.go [hunk 1]
```diff
@@ -74,6 +74,50 @@ func (im *LegacyIBCModule) OnChanOpenInit(
 	return reconstructVersion(im.cbs, negotiatedVersions)
 }
 
+// OnChanOpenTry implements the IBCModule interface.
+func (im *LegacyIBCModule) OnChanOpenTry(
```
Comment (chatton) [inline modules/core/05-port/types/ibc_legacy_module.go:78]: this is almost an exact duplicate, I played around with trying to generalize the Init/Try functions, but it ended up looking even uglier, happy to revisit this idea, but I think it's fine as is for now unless someone has a suggestion for removing the duplication.

File: modules/apps/29-fee/ibc_middleware.go [hunk 2]
```diff
@@ -70,33 +70,11 @@ func (im IBCMiddleware) OnChanOpenTry(
 	counterparty channeltypes.Counterparty,
 	counterpartyVersion string,
 ) (string, error) {
-	versionMetadata, err := types.MetadataFromVersion(counterpartyVersion)
-	if err != nil {
-		// Since it is valid for fee version to not be specified, the above middleware version may be for a middleware
-		// lower down in the stack. Thus, if it is not a fee version we pass the entire version string onto the underlying
-		// application.
-		return im.app.OnChanOpenTry(ctx, order, connectionHops, portID, channelID, counterparty, counterpartyVersion)
+	if strings.TrimSpace(counterpartyVersion) != "" && counterpartyVersion != types.Version {
```
Comment (colin-axner) [inline modules/apps/29-fee/ibc_middleware.go:73]:
```suggestion
	if counterpartyVersion != types.Version {
```

Comment (colin-axner) [reply modules/apps/29-fee/ibc_middleware.go:73]: I think on Try, the version must be set

Comment (chatton) [reply modules/apps/29-fee/ibc_middleware.go:73]: makes sense!

**Pull request merged**